### PR TITLE
Structured data for degree qualifications

### PR DESCRIPTION
**Pull request by paulrobertlloyd**

### CODE REVIEW

**Comment by fofr** [comment]
What's the need around knowing when they started their degree?

**Comment by fofr** [comment]
Can this entry indicate where the data comes from (or will come from)?

Not sure if the data is meant to be complete.

My undergraduate degree was a masters:
Screen Shot 2020-04-21 at 09 32 34

File: app/posts/apply-for-teacher-training/2020-04-16-structured-data-for-degrees.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    caption: "The candidate can enter free text, but suggestions from HESA’s list of degree establishments are provided."
+  }, {
+    text: "What grade is your degree? (Undergraduate degree)",
+    caption: "Undergraduate degree grades are provided, but candidates can choose and specify ‘Other’"
+  }, {
+    text: "What grade is your degree? (Undergraduate degree) – Other",
+    caption: "If ‘Other’ is choosen, we allow a free text answer, but suggest other known (but little used) undergraduate degree types."
+  }, {
+    text: "What grade is your degree? (Postgraduate degree)",
+    caption: "Postgraduate degree grades are provided, but candidates can choose ‘Unknown’ or specify ‘Other’"
```
Comment (fofr) [inline app/posts/apply-for-teacher-training/2020-04-16-structured-data-for-degrees.md:66]: My undergraduate masters had the same grading system as an undergraduate degree. (First, 2:1, etc)

Comment (fofr) [reply app/posts/apply-for-teacher-training/2020-04-16-structured-data-for-degrees.md:66]: Typo: Distinction, not destinction

File: app/posts/apply-for-teacher-training/2020-04-16-structured-data-for-degrees.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    caption: "The candidate can enter free text, but suggestions from HESA’s list of degree subjects are provided."
+  }, {
+    text: "What institution did you go to?",
+    caption: "The candidate can enter free text, but suggestions from HESA’s list of degree establishments are provided."
+  }, {
+    text: "What grade is your degree? (Undergraduate degree)",
+    caption: "Undergraduate degree grades are provided, but candidates can choose and specify ‘Other’"
+  }, {
+    text: "What grade is your degree? (Undergraduate degree) – Other",
+    caption: "If ‘Other’ is choosen, we allow a free text answer, but suggest other known (but little used) undergraduate degree types."
```
Comment (fofr) [inline app/posts/apply-for-teacher-training/2020-04-16-structured-data-for-degrees.md:63]: Typo: Chosen, not choosen

**Comment by fofr** [comment]
When no suggestions are found, I don't think it will always be clear that a user can continue with their free text, rather than having to make a selection.

I wonder if the empty state text in the autocomplete could be more helpful.

**Comment by fofr** [comment]
How would you enter two different undergraduate degrees? (A friend of mine took a 3yr undergrad computer science course at Cambridge, followed by a 3yr undergrad law course at UCL)

**Comment by fofr** [comment]
Suggestion: We ask if degree is undergrad or postgrad when adding, rather than inferring. We add validation if they don't add an undergrad degree when marking section as complete/reviewing.

**Comment by paulrobertlloyd** [comment]
> How would you enter two different undergraduate degrees? (A friend of mine took a 3yr undergrad computer science course at Cambridge, followed by a 3yr undergrad law course at UCL)

That’s possible; it’s just that we force the first degree to be an undergraduate degree (by limiting the degree type options, and using ‘undergraduate’ in the title). Subsequent degrees can be undergraduate too. Do you think that’s not clear?

**Comment by paulrobertlloyd** [comment]
> Suggestion: We ask if degree is undergrad or postgrad when adding, rather than inferring. We add validation if they don't add an undergrad degree when marking section as complete/reviewing.

I wondered about that, but my thought that degree type was a strong enough indicator. I guess that falls apart if your degree type is not shown and/or you are able to enter your own type.

**Comment by paulrobertlloyd** [comment]
> When no suggestions are found, I don't think it will always be clear that a user can continue with their free text, rather than having to make a selection.
>
> I wonder if the empty state text in the autocomplete could be more helpful.

I can try this, but I think providing text when there are no options prevents another answer being submitted. But I may be wrong.

**Comment by fofr** [comment]
> my thought that degree type was a strong enough indicator

How do you infer if a Masters is undergrad or postgrad?

**Comment by paulrobertlloyd** [comment]
> Can this entry indicate where the data comes from (or will come from)?

Links to HESA datasets are in the related items. Maybe these links should be contextual, too?

**Comment by paulrobertlloyd** [comment]
MEng is in the dataset: https://www.hesa.ac.uk/collection/c20053/e/degtype (Code 211), and in the data in the prototype.

I wonder if the autocomplete is case sensitive?

**Comment by paulrobertlloyd** [comment]
Perhaps we shouldn’t limit the type options for the first degree? Or only omit doctorates?

**Pull request merged**